fortex/health/processors/coreference_processor.py

```diff
@@ -0,0 +1,209 @@
+
+                    group = CoreferenceGroup(input_pack)
+                    group.add_members(mentions)
+
+                    article.coref_groups.append(group)
+
+    @classmethod
+    def default_configs(cls):
+        r"""
+        This defines a basic config structure for `CoreferenceProcessor`.
+
+        Following are the keys for this dictionary:
+         - `entry_type`: Input entry type. Default: `"ft.onto.base_ontology.Document"`.
+         - `mention_type`: Output mention type.
+            Default: `"ftx.medical.clinical_ontology.MedicalEntityMention"`.
+            It can also be set to `"ft.onto.base_ontology.EntityMention"`.
+         - `model`: the neural net model to be used by NeuralCoref. If set to `True`,
+            a new instance will be created with `NeuralCoref.Model()`. Default: `True`.
+            in `NeuralCoref.from_disk()` or `NeuralCoref.from_bytes()`.
+         - `greedyness` (`float`): A number between 0 and 1 determining how greedy
+            the model is about making coreference decisions
+            (more greedy means more coreference links). Default: `0.5`.
+         - `max_dist` (`int`): How many mentions back to look when considering possible
+            antecedents of the current mention. Decreasing the value will cause
+            the system to run faster but less accurately. Default: `50`.
+         - `max_dist_match` (`int`): The system will consider linking the current mention
+            to a preceding one further than max_dist away if they share a noun or
+            proper noun. In this case, it looks max_dist_match away instead. Default: `500`.
+         - `blacklist` (`bool`): Should the system resolve coreferences for pronouns in the
+            following list: ["i", "me", "my", "you", "your"]. Default `True`.
+         - `store_scores` (`bool`): Should the system store the scores for the coreferences
+            in annotations. Default: `True`
+         - `conv_dict` (`dict(str, list(str))`): A conversion dictionary that you can use
+            to replace the embeddings of rare words (keys) by an average of the embeddings
+             of a list of common words (values). Ex: `conv_dict={"Angela": ["woman", "girl"]}`
+             will help resolving coreferences for Angela by using the embeddings for the more
+             common woman and girl instead of the embedding of Angela.
+             This currently only works for single words (not for words groups). Default: `None`.
+
+        Returns: A dictionary with the default config for this processor.
+        """
+        return {
+            "entry_type": "ft.onto.base_ontology.Document",
+            "mention_type": "ftx.medical.clinical_ontology.MedicalEntityMention",
+            "lang": "en_core_web_sm",
+            "model": True,
+            "greedyness": 0.5,
+            "max_dist": 50,
+            "max_dist_match": 500,
+            "blacklist": True,
+            "store_scores": True,
```

Comment:
I didn't find that `store_scores` affect our output. If not, we shouldn't expose these additional features.

Reply:
It doesn't affect. I keep this because if some day `CoreferenceGroup` need to record confidence, then it will be useful. Accoring to the comments you left, we need better readability, so I will remove it.

Reply:
Well let's not hope for "some day", when a piece of code is pushed, it will stay there for a long time and no one will touch it.

Note that it is perfectly fine to include `store_scores` if we actually store them in Datapack, but we didn't. The thing about readability is that the options should make sense and have an effect in a way that the user expects. Having a long list of parameters is not the worst problem in readability, but an option without effect and documentation would impact readability. Imagine a user asking "I set `store_scores` to True but where is my score?"

Reply:
Removed store_scores.